packages/sanity/src/core/config/document/inspector.ts

```diff
@@ -1,26 +1,30 @@
-import {ValidationMarker} from '@sanity/types'
 import {ButtonTone} from '@sanity/ui'
 import {ComponentType} from 'react'
 
 /** @beta */
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
 export interface DocumentInspectorProps {
+  documentId: string
+  documentType: string
   onClose: () => void
 }
 
 /** @beta */
 export type DocumentInspectorComponent = ComponentType<DocumentInspectorProps>
 
 /** @beta */
-export interface DocumentInspectorCallbackContext {
-  validation: ValidationMarker[]
+export interface DocumentInspectorMenuItemContext {
+  documentId: string
+  documentType: string
 }
 
 /** @beta */
 export interface DocumentInspectorMenuItem {
-  hidden?: boolean | ((props: DocumentInspectorCallbackContext) => boolean)
+  hidden?: boolean
   hotkeys?: string[]
   icon?: ComponentType
+  showAsAction?: boolean
```

Comment:
Consider making this a string instead of a boolean to make it more configurable in the future?

`location: 'default' | 'menu' | ...`